File: octopoints-kotlin/app/src/main/java/dev/tpavani/octopoints/db/dao/JoinTeamDao.kt
package dev.tpavani.octopoints.db.dao

import androidx.room.Dao
import androidx.room.Query
import dev.tpavani.octopoints.db.entity.JoinTeamEntity
import dev.tpavani.octopoints.db.entity.UserEntity

@Dao
interface JoinTeamDao : IDao<JoinTeamEntity> {
    @Query("SELECT U.* FROM users U, jointeam J WHERE J.teamId=:teamId AND J.userId=U.id")
    suspend fun getTeammates(teamId: Int): List<UserEntity>
}

File: octopoints_flutter/lib/db/dao/JoinTeamDao.dart
import 'package:floor_annotation/floor_annotation.dart';
import 'package:octopoints_flutter/db/dao/IDao.dart';
import 'package:octopoints_flutter/db/entity/JoinTeamEntity.dart';
import 'package:octopoints_flutter/db/entity/UserEntity.dart';

@dao
abstract class JoinTeamDao extends IDao<JoinTeamEntity> {
  @Query(
      'SELECT U.* FROM users U, jointeam J WHERE J.teamId=:teamId AND J.userId=U.id')
  Future<List<UserEntity>> getTeammates(int teamId);
}
